lis3dh/lis3dh.go

```diff
@@ -0,0 +1,133 @@
+func (d *Device) Configure() {
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL5, []byte{0x80})
+	time.Sleep(5 * time.Millisecond) // reset takes 5ms
+
+	// enable all axes, normal mode
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL1, []byte{0x07})
+
+	// 400Hz rate
+	d.SetDataRate(LIS3DH_DATARATE_400_HZ)
+
+	// High res & BDU enabled
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL4, []byte{0x88})
+
+	// DRDY on INT1
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL3, []byte{0x10})
+
+	// enable ADC
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_TEMPCFG, []byte{0x80})
+}
+
+// Connected returns whether a LIS3DH has been found.
+// It does a "who am I" request and checks the response.
+func (d *Device) Connected() bool {
+	data := []byte{0}
+	err := d.bus.ReadRegister(uint8(d.Address), WHO_AM_I, data)
+	if err != nil {
+		println(err.Error())
+	}
+	return data[0] == 0x33
+}
+
+// SetDataRate sets the speed of data collected by the LIS3DH.
+func (d *Device) SetDataRate(rate DataRate) {
+	ctl1 := []byte{0}
+	err := d.bus.ReadRegister(uint8(d.Address), LIS3DH_REG_CTRL1, ctl1)
+	if err != nil {
+		println(err.Error())
+	}
+	// mask off bits
+	ctl1[0] &^= 0xf0
+	ctl1[0] |= (byte(rate) << 4)
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL1, ctl1)
+}
+
+// SetRange sets the G range for LIS3DH.
+func (d *Device) SetRange(r Range) {
+	ctl := []byte{0}
+	err := d.bus.ReadRegister(uint8(d.Address), LIS3DH_REG_CTRL4, ctl)
+	if err != nil {
+		println(err.Error())
+	}
+	// mask off bits
+	ctl[0] &^= 0x30
+	ctl[0] |= (byte(r) << 4)
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL4, ctl)
+}
+
+// GetRange returns the current G range for LIS3DH.
+func (d *Device) GetRange() (r Range) {
+	ctl := []byte{0}
+	err := d.bus.ReadRegister(uint8(d.Address), LIS3DH_REG_CTRL4, ctl)
+	if err != nil {
+		println(err.Error())
+	}
+	// mask off bits
+	r = Range(ctl[0] >> 4)
+	r &= 0x03
+
+	return r
+}
+
+// Update reads and stores the sensor values.
+func (d *Device) Update() {
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_OUT_X_L|0x80, nil)
+
+	data := []byte{0, 0, 0, 0, 0, 0}
+	d.bus.Tx(d.Address, nil, data)
+
+	buf := bytes.NewReader(data)
+	binary.Read(buf, binary.LittleEndian, &(d.x))
+	binary.Read(buf, binary.LittleEndian, &(d.y))
+	binary.Read(buf, binary.LittleEndian, &(d.z))
+}
+
+// Acceleration returns the adjusted x, y and z axis in milli-Gs.
+func (d *Device) Acceleration() (x int16, y int16, z int16) {
+	r := d.GetRange()
+	divider := float32(1)
+	switch r {
+	case LIS3DH_RANGE_16_G:
+		divider = 1365
+	case LIS3DH_RANGE_8_G:
+		divider = 4096
+	case LIS3DH_RANGE_4_G:
+		divider = 8190
+	case LIS3DH_RANGE_2_G:
+		divider = 16380
+	}
+
+	return int16(float32(d.x) / divider * 1000), int16(float32(d.y) / divider * 1000), int16(float32(d.z) / divider * 1000)
+}
+
+// RawXYZ returns the raw x, y and z axis from the LIS3DH
+func (d *Device) RawXYZ() (x int16, y int16, z int16) {
```

Comment:
Why `RawXYZ`? To me, `RawAcceleration` seems more appropriate (or if you're going for `ReadAcceleration`, it would be `ReadRawAcceleration`).